exercises/practice/leap/.approaches/ternary-operator/content.md

```diff
@@ -0,0 +1,34 @@
+# Ternary operator
+
+```python
+def leap_year(year):
+    return (not year % 400 if not year % 100 else not year % 4)
```

Comment:
I'd remove the parens here, unless you wan to break the ternary up for readability like:

```python

return (not year % 400 
        if not year % 100 
        else not year % 4)
```

Otherwise, I think they're confusing, especially given their use in the boolean expressions elsewhere.

Reply:
What do you think of [this site](https://www.tutorialspoint.com/online_python_formatter.htm) for formatting Python? I ask, because it put the parens around there which I did not put originally. If you can recommend a better online formatter, I will gladly use it, because I agree, I like it better without the parens.

Reply:
Whelp.  They certainly have their own style.  Dunno that it's one I'd tell students to entirely follow.

So.  I run the risk here of contradicting myself.  I (mostly) resent [Black](https://github.com/psf/black) as a formatter - _especially_ when it comes to nested data structures.  However, it's on its way to becoming somewhat of a standard in many places and projects.  And [this is what the online playground did to the code](https://black.vercel.app/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AE7AIhdAD2IimZxl1N_Wk6rkcNtQDTqPhpt4m4s-Ya1u-8_zvePRI8_3FGtyS4fs63JHUuz20LxVZ1l865b7C6g8jaHL9X-3bNNoNw_6EA35yjWSDUN7nJ0YtPVsYyocZQb7zWoa2sfR6QivKAiPSHJVVsPFCpzfuxJ3SOPe0a_4XsLf99Qi_qyg6qjvAAAX9YCvn3tDi0AAaQBvAIAANmLmCixxGf7AgAAAAAEWVo=).  We use Black here in the repo to auto-format all of our generated problem spec test files.

**BUT**

If you are going to go the Black route, I would ask that you put skips in for data structures.  It really does a number on things like `dicts` |`tuples` | `lists` etc. - especially when they're nested.  It also has a restricted line length (90 as opposed to our 120) - but if you are doing mostly snippets, that may not matter.

Google uses [yapf](https://github.com/google/yapf).  We have a [yapf config](https://github.com/exercism/python/blob/main/.style.yapf) here in the repo, and I (mostly) prefer that to Black's, and that's the one we've pointed to in our contributor docs. That being said, [yapf online](https://yapf-online--mat1.repl.co/) (_or at least the online implementation I found_) didn't _remove_ the parens when they were present.  

![image](https://user-images.githubusercontent.com/5923094/202594278-49e28a6f-6d20-4eae-9aab-ab8d01b86c4c.png)


But it also didn't _add_ any. - I am not so sure that that online demo is "official".    When I have time later, I will run things locally on my machine and double check behavior based on our repo config.

Reply:
Okay, thanks, I'll use yapf online going forward. For now I'll gladly remove the parens, as it was the other site that added them to my original parentheses-less code.